### PR TITLE
Make core game engine stateless and integrate with the online application

### PR DESCRIPTION
**Pull request by BelfDev**
# Stateless Core Game Logic and Online App Integration

- Extract state from the core game engine
- Eliminate the observer pattern; CommandLineController handles the CLI state imperatively
- Create the online App REST API
- Create all the request/response mappings and integrate with domain models
- Review the game logic and fix spotted mistakes
- Resolve development merge conflicts
- Enhance performance in general

### CODE REVIEW

File: src/main/java/com/toptrumps/cli/CommandLineController.java [hunk 1]
```diff
@@ -3,109 +3,215 @@
...
 
-        if (input.equalsIgnoreCase("s")) {
+        scanner.nextLine();
+    }
+
+    private void startNewGame(int numberOfOpponents) {
+        System.out.println("Game Start");
+
+        List<Card> communalPile = new ArrayList<>();
+        int rounderNumber = 0;
```
Comment (rjs68) [inline src/main/java/com/toptrumps/cli/CommandLineController.java]: Very minor but should this be round rather than rounder?

**Review by nathanieldschneider — APPROVED** [review]
nice wan

**Review by rjs68 — APPROVED** [review]
Only one minor comment about the rounderNumber name but I don't wanna delay things by requesting changes

File: src/main/java/com/toptrumps/cli/CommandLineController.java [hunk 2]
```diff
@@ -114,63 +220,31 @@ private void showRoundResult(RoundOutcome outcome){
...
 
-    @Override
-    public void onGameOver(Player winner) {
-        String message = String.format("\nGAME OVER, %s won\n\n", winner.getName());
-        System.out.println(message);
-        scanner.nextLine(); //clear the scanner ready for new game selection
-        newGameHandler();
-    }
-
-    private void numberOfHumanPlayerCards(int cardsRemaining ) {
```
Comment (hughwin) [inline src/main/java/com/toptrumps/cli/CommandLineController.java:145]: Why was this deleted? It was kinda useful.

Comment (BelfDev) [reply src/main/java/com/toptrumps/cli/CommandLineController.java:145]: ```newGameHandler()``` has been refactored and now is part of ```start()```; I've also restored its call inside ```onGameOver```. I suggest we incorporate an option to simply **quit** the game.

File: src/main/java/com/toptrumps/cli/CommandLineController.java (hunk 1 above)
Comment (BelfDev) [reply src/main/java/com/toptrumps/cli/CommandLineController.java]: Hahaha thanks! Sorry, this was a mistake indeed.

**Review by hughwin — APPROVED** [review]
